arctic/store/_pandas_ndarray_store.py

```diff
@@ -183,7 +183,7 @@ def can_write_type(data):
 
     def can_write(self, version, symbol, data):
         if self.can_write_type(data):
-            if np.any(data.dtypes.values == 'object'):
+            if np.any(data.dtypes.values == 'object') or data.index.dtype.hasobject:
```

Comment:
Should this and below be hasobject also?

Reply:
"hasobject" can be applied only on a per-dtype basis:

```
In[13]: mydf.dtypes.hasobject
Traceback (most recent call last):
  File "/dev/shm/dpediaditakis/pyenvs/fedex.2.7.15/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-13-377951dabaaf>", line 1, in <module>
    mydf.dtypes.hasobject
  File "/dev/shm/dpediaditakis/pyenvs/fedex.2.7.15/lib/python2.7/site-packages/pandas/core/generic.py", line 4376, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'hasobject'
```

An alternative would be:
```
In[14]: np.any([dt.hasobject for dt in mydf.dtypes])
Out[14]: False
```

but I'd rather cause as less regression risks as possible at this stage

Reply:
@yschimke  I liked your comment/suggestion, so now the code has homogeneous check for Series object dtype, and added more tests to cover this